sagan-common/src/main/java/sagan/projects/ProjectRelease.java

```diff
@@ -183,44 +187,75 @@ public ProjectRelease createWithVersionPattern() {
         return release;
     }
 
+    private static final Map<String, Integer> SPECIAL_MEANINGS =
+            ImmutableMap.of("dev", -1, "rc", 1, "release", 2, "final", 3);
+
```

Comment:
This is the logic that Gradle/Maven/Ivy consider special meanings... the rest of the possibilities fall out in the wash based on lexicographical ordering